File: 2.Dagger/Subcomponents/AnnotationParameter/src/main/java/com/app/annotationparameter/fragment/MainFragment.java
package com.app.annotationparameter.fragment;

import android.content.Context;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.app.annotationparameter.R;
import com.app.annotationparameter.activity.MainActivity;

import javax.inject.Inject;
import javax.inject.Named;

public class MainFragment extends Fragment {
    @Inject
    @Named("app string")
    String appString;

    @Inject
    @Named("activity string")
    String activityString;

    @Inject
    @Named("fragment string")
    String fragmentString;


    public MainFragment() {}

    @Override
    public void onAttach(@NonNull Context context) {
        ((MainActivity)requireActivity()).fragmentComponentFactory.create().inject(this);
        super.onAttach(context);
    }

    // TODO: Rename and change types and number of parameters
    public static MainFragment newInstance(String param1, String param2) {
        MainFragment fragment = new MainFragment();
        return fragment;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.i(TAG, appString + ", " + activityString + ", " +fragmentString);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view =  inflater.inflate(R.layout.fragment_main, container, false);

        return view;
    }

    private static final String TAG = "MainFragment";
}

File: 2.Dagger/Hilt/app/src/main/java/com/app/hilt/fragment/MainFragment.kt
package com.app.hilt.fragment

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.app.hilt.R
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

@AndroidEntryPoint
class MainFragment : Fragment() {

    @FragmentString
    @Inject lateinit var fragmentStr:String

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {

        // Inflate the layout for this fragment
        val view = inflater.inflate(R.layout.fragment_main, container, false)
        return view
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        Log.i(TAG, fragmentStr)
    }

    companion object {
        private const val TAG = "MainFragment"
    }

}